ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1091 @@
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(Init))]",
+                "\npublic interface ReportScope : IMutableScope<string>, ReportingNodeFormsEntity, MonthlyPeriodFormsEntity, ScenarioFormsEntity, CurrencyFormsEntity, ",
+                "\n                               BasicSliceAndDiceRowsFormsEntity, BasicSliceAndDiceColumnsFormsEntity, BasicFilterFormsEntity {",
+                "\n    protected IPivotFactory report => GetStorage().Report;",
+                "\n    protected IExportVariable export => GetStorage().Export;",
+                "\n    protected int headerColumnWidthValue => 250;",
+                "\n",
+                "\n    HashSet<(ReportIdentity, CurrencyType)> GetDataIdentities() => GetStorage().GetIdentities((Year, Month), ReportingNode, Scenario, CurrencyType); // TODO, add filter for identities, if the property is exposed at this level",
+                "\n",
+                "\n    IDataCube<ReportVariable> GetData() => default;",
+                "\n",
+                "\n    async Task<GridOptions> ToReportAsync() {",
+                "\n        await GetStorage().InitializeAsync((Year, Month), ReportingNode, Scenario, CurrencyType);",
+                "\n        var dataScope = GetScope<Data>(((Year, Month), ReportingNode, Scenario, CurrencyType, dataFilter));",
+                "\n        dataScope.InputDataCube = GetData();",
+                "\n        return await GetReportTaskAsync(dataScope.DataCube);",
+                "\n    }",
+                "\n",
+                "\n    // Using this routine is highly discouraged due to the mutlithreading issue -A.K.",
+                "\n    // Avoid using these methods if working with the DB -- it will trigger synchronization error in access to the DB",
+                "\n    async Task<ExportResult> ToCsvAsync(string fileName){",
+                "\n        await GetStorage().InitializeAsync((Year, Month), ReportingNode, Scenario, CurrencyType);",
+                "\n        var dataScope = GetScope<Data>(((Year, Month), ReportingNode, Scenario, CurrencyType, dataFilter));",
+                "\n        dataScope.InputDataCube = GetData();",
+                "\n        return await export.ToCsv(fileName)",
+                "\n                        .ForDataCube(dataScope.DataCube, config => config.WithQuerySource(workspace)",
+                "\n                                                                        .SliceRowsBy(rowSlices)",
+                "\n                                                                        .SliceColumnsBy(columnSlices))",
+                "\n                        .ExecuteAsync();",
+                "\n    }",
+                "\n",
+                "\n    async Task<ExportResult> ToExcelAsync(string fileName){",
```

Comment:
@andrey-katz-systemorph  should these export methods be commented out?

Reply:
They could be. I left them here, since they are OK, as long as one works without a DB. It would be your call whether to comment them out or not, I do not have any strong opinion.

Reply:
As discussed with @dcolleoni we leave it to be used in projects with no DB.